files/default/plugins/printers.rb

```diff
@@ -0,0 +1,44 @@
+provides 'ohai_gecos'
+
+if ohai_gecos.nil?
+  ohai_gecos Mash.new
+end
+
+begin
+  entries = []
+  printers = []
+
+  entries = Mixlib::ShellOut.new("lpstat -a | awk {'print $1'}")
```

Comment:
In case of a printer not being ready (accepting jobs), lpstat -a outputs an additional diagnose line; thus, the output of lpstat -a | awk {'print $1'} can include lines that are not printer names. 
Diagnose lines are indented and could be skipped by detecting the leading blank.
lpstat -v outputs no diagnose, but is not as simple for parsing; leading words are language dependent.